charts/prefect-worker/templates/_helpers.tpl

```diff
@@ -9,15 +9,35 @@ Create the name of the service account to use
 {{- end -}}
 {{- end -}}
 
+{{/*
+Require Prefect Cloud Account ID
+*/}}
+{{- define "cloud.requiredConfig.accountId" -}}
+{{- if eq .Values.worker.apiConfig "cloud" }}
+    {{- required "A Prefect Cloud Account ID is required (worker.cloudApiConfig.accountId)" .Values.worker.cloudApiConfig.accountId | printf "%s"  -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Require Prefect Cloud Workspace ID
+*/}}
+{{- define "cloud.requiredConfig.workspaceId" -}}
+{{- if eq .Values.worker.apiConfig "cloud" }}
+    {{- required "A Prefect Cloud Workspace ID is required (worker.cloudApiConfig.workspaceId)" .Values.worker.cloudApiConfig.workspaceId | printf "%s"  -}}
+{{- end -}}
+{{- end -}}
+
 {{/*
   worker.apiUrl:
     Define API URL for cloud or server worker install
 */}}
 {{- define "worker.apiUrl" -}}
 {{- if eq .Values.worker.apiConfig "cloud" }}
-{{- printf "%s/accounts/%s/workspaces/%s" .Values.worker.cloudApiConfig.cloudUrl .Values.worker.cloudApiConfig.accountId .Values.worker.cloudApiConfig.workspaceId | quote }}
+    {{- printf "%s/accounts/%s/workspaces/%s" .Values.worker.cloudApiConfig.cloudUrl (include "cloud.requiredConfig.accountId" .) (include "cloud.requiredConfig.workspaceId" .) | quote }}
+{{- else if and (eq .Values.worker.apiConfig "server") .Values.worker.serverConfig.apiUrl.fullyQualifiedEndpoint }}
+    {{- printf .Values.worker.serverConfig.apiUrl.fullyQualifiedEndpoint | quote }}
 {{- else }}
-{{- .Values.worker.serverApiConfig.apiUrl | quote }}
+    {{- printf "http://%s.%s.svc.cluster.local:%s/api" .Values.worker.serverConfig.apiUrl.prefectServer.serviceName .Values.worker.serverConfig.apiUrl.prefectServer.namespace .Values.worker.serverConfig.apiUrl.prefectServer.servicePort | quote}}
```

Comment:
my opinion - i think the operator should just construct this k8s endpoint themselves, vs us handling this third case and having more value inputs overall.  so sharing the `fullyQualifiedEndpoint` input for both self-hosted or internal k8s